homeassistant/components/alexa/capabilities.py

```diff
@@ -953,6 +954,10 @@ def get_property(self, name):
             ):
                 return f"{cover.ATTR_POSITION}.{mode}"
 
+        if self.instance == f"{light.DOMAIN}.effect":
+            mode = self.entity.attributes.get(light.ATTR_EFFECT)
```

Comment:
```suggestion
            mode = self.entity.attributes.get(light.ATTR_EFFECT)
            if mode in (
                light.EFFECT_COLORLOOP,
                light.EFFECT_RANDOM,
                light.EFFECT_WHITE,
            ):
```
We should check for known constants. The light component currently only supports the following effect constants.

Reply:
Not all lights have those effects, Tasmota for example has 0 to 4. We could generate something here using the lights effects_list attribute

Reply:
What's the resulting utterances for a Tasmota effect with 0 to 4? 

> Alexa, set Tasmota to 0?
> Alexa, set Tasmota to zero?

That could be confusing to the voice model when a `PercentageController`, `PowerLevelController`, or `RangeController'. is also yielded.

I'm not against supporting custom lists, but we have to account for any/all mode names that are provided to Alexa, translate them to an actual word for phrase, and provide synonyms when possible.

Reply:
It would be: "Alexa, set effect on Tasmota to 0" or "Alexa, set effect on Tasmota to zero"  both worked when tested and converted it back to "0" as the mode value.

It would be nice if Tasmota didn't use such undescriptive names but I don't think this is a HA problem. However.... it would be nice if we could do "Alexa set effect on Tasmota to <script name>" where script name is some arbitrary script that takes light tasmota and does some actions on it - this way we could create our own effects in HA or alias affects via scripts.

Reply:
Out of scope ;-)